File: data/src/main/java/com/everis/data/network/firebase/FireBaseAuth.java
package com.everis.data.network.firebase;

import com.google.firebase.auth.FirebaseUser;

import rx.Observable;

/**
 * Created by everis on 25/04/18.
 */

public interface FireBaseAuth {

    Observable<FirebaseUser> createUser(String emaail, String password);
    Observable<FirebaseUser> signInUser(String emaail, String password);
    Observable<FirebaseUser> observeAuthStateChange();
    Observable<Void> logOut();
    Observable<Void> deleteUser();
    Observable<FirebaseUser> checkUserLoggedIn();

}

File: data/src/main/java/com/everis/data/network/firebase/FireBaseAuth.kt
package com.everis.data.network.firebase

import com.google.firebase.auth.FirebaseUser

import rx.Observable


interface FireBaseAuth {

    fun createUser(email: String, password: String): Observable<FirebaseUser>
    fun signInUser(email: String, password: String): Observable<FirebaseUser>
    fun observeAuthStateChange(): Observable<FirebaseUser>
    fun logOut(): Observable<Void>
    fun deleteUser(): Observable<Void>
    fun checkUserLoggedIn(): Observable<FirebaseUser>

}
